src/realm.js

```diff
@@ -1737,12 +1745,29 @@ export class Realm {
     return !this._abstractValuesDefined.has(nameString);
   }
 
-  getTemporalBuildNodeEntryArgsFromDerivedValue(value: Value): void | TemporalBuildNodeEntryArgs {
+  getTemporalBuildNodeEntryFromDerivedValue(value: Value): void | TemporalBuildNodeEntry {
     let name = value.intrinsicName;
-    invariant(name);
-    let preludeGenerator = this.preludeGenerator;
-    invariant(preludeGenerator !== undefined);
-    let temporalBuildNodeEntryArgs = preludeGenerator.derivedIds.get(name);
-    return temporalBuildNodeEntryArgs;
+    if ((value instanceof AbstractValue && value.kind === "conditional") || !name) {
```

Comment:
This conditional check is still here. This tells me something is wrong somewhere.

Reply:
Okay I will remove this check and tell you what test fails.

Reply:
This test fails:

```js
function f(x, foo, bar) {
  var a = Object.assign({}, foo, bar, {a: 1});
  foo = {};
  var b
  if (x) {
    b = Object.assign({}, a, {a: 2});
  } else {
    b = Object.assign({}, a, {a: 5});
  }
  var c = Object.assign({}, b, {a: 2}, {d: 5});
  return c;
}
```

Because `b` is a `conditional` abstract object value that has a temporal entry.

Reply:
Sure, `b` becomes a conditional, but it does not become a derived id, and it does not have an intrinsic name. So the `!name` check should be sufficient.

Reply:
Got you now. Applied change.